tooling/camel-karaf-feature-maven-plugin/src/main/java/org/apache/camel/karaf/feature/maven/EnsureWrapBundleVersionMojo.java

```diff
@@ -0,0 +1,319 @@
+    private String featuresFilePath;
+
+    @Parameter(property = "targetFeature", required = false)
+    private String targetFeature = null;
+
+    public String getFeaturesFilePath() {
+        return featuresFilePath;
+    }
+
+    public void setFeaturesFilePath(String featuresFilePath) {
+        this.featuresFilePath = featuresFilePath;
+    }
+
+    public String getTargetFeature() {
+        return targetFeature;
+    }
+
+    public void setTargetFeature(String targetFeature) {
+        this.targetFeature = targetFeature;
+    }
+
+    @Override
+    public void execute() throws MojoExecutionException {
+        Features featuresData = JaxbUtil.unmarshal(getFeaturesFilePath(), false);
+        List<Feature> features = featuresData.getFeature();
+
+        if (getTargetFeature() != null) {
+            boolean featureFound = false;
+            for (Feature feature : features) {
+                // all the feature versions will be modified
+                if (getTargetFeature().equals(feature.getName())) {
+                    featureFound = true;
+                    processFeature(feature);
+                }
+            }
+            if (!featureFound) {
+                getLog().warn("Feature %s not found. File '%s' wasn't modified".formatted(getTargetFeature(),
+                        getFeaturesFilePath()));
+                return;
+            }
+        } else {
+            processFeatures(features);
+        }
+
+        marshal(featuresData);
+    }
+
+    private void marshal(Features featuresData) throws MojoExecutionException {
+        try (StringWriter writer = new StringWriter()) {
+            JaxbUtil.marshal(featuresData, writer);
+
+            String result = writer.toString().replace(DEFAULT_HEADER, LICENCE_HEADER);
+
+            Path path = Paths.get(getFeaturesFilePath().replaceFirst(FILE_PROTOCOL, ""));
+            Files.writeString(path, result);
+
+            getLog().info("File '%s' was successfully modified and saved".formatted(getFeaturesFilePath()));
+        } catch (Exception e) {
+            getLog().error("File '%s' was successfully modified but an error occurred while saving it"
+                    .formatted(getFeaturesFilePath()), e);
+            throw new MojoExecutionException(e);
+        }
+    }
+
+    private void processFeatures(List<Feature> features) {
+        for (Feature feature : features) {
+            processFeature(feature);
+        }
+    }
+
+    private void processFeature(Feature feature) {
+        for (Bundle bundle : feature.getBundle()) {
+            String location = bundle.getLocation();
+            if (location != null && location.startsWith(WRAP_PROTOCOL)) {
+                try {
+                    bundle.setLocation(processLocation(location));
+                } catch (Exception e) {
+                    getLog().error("Bundle location '%s' was ignored: %s".formatted(location, e.getMessage()), e);
```

Comment:
The error message should rather be something like "Could not process the Bundle location '%s': '%s'"

Reply:
Done 👍